### PR TITLE
[SG-232 & SG-251] Fix color issues with organization badge

### PR DESCRIPTION
**Pull request by differsthecat**
## Type of change

- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

- Fix text color tending to be too dark for organization badge
- Make organization badge use avatar color for personal vault items

This will be cherry picked to rc

## Code changes

- **src/app/modules/vault/modules/organization-badge/organization-name-badge.component.ts:** 
  - Take an input with the profile name to use to calculate the badge color for personal vault items
  - Adjust calculation of text color (still using one of the recommended approaches)
- **src/app/organizations/vault/ciphers.component.ts:**
  - Get profile name to use for color calculation
## Screenshots
Screen Shot 2022-05-09 at 1 59 17 PM

## Testing requirements

- Make sure the badge for individual vault items uses the avatar color
- Make sure text color properly contrasts with badge color

## Before you submit

- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: src/app/vault/ciphers.component.ts [hunk 1]
```diff
@@ -65,6 +68,8 @@ export class CiphersComponent extends BaseCiphersComponent implements OnDestroy
   // Do not use ngOnInit() for anything that requires sync data.
   async load(filter: (cipher: CipherView) => boolean = null, deleted = false) {
     await super.load(filter, deleted);
+    const profile = await this.apiService.getProfile();
```
Comment (Hinton) [inline src/app/vault/ciphers.component.ts:71]: Do we want to do a http post here?

**Review by Hinton — CHANGES_REQUESTED** [review]
There must be a better way to get the current accounts name than fetching it on every page load?

File: src/app/vault/ciphers.component.ts (hunk 1 above)
Comment (differsthecat) [reply src/app/vault/ciphers.component.ts]: As per your suggestion I used tokenService instead, thank you!

File: src/app/modules/vault/modules/organization-badge/organization-name-badge.component.ts [hunk 2]
```diff
@@ -49,18 +50,7 @@ export class OrganizationNameBadgeComponent implements OnInit {
...
-      if (c <= 0.03928) {
-        return c / 12.92;
-      } else {
-        return Math.pow((c + 0.055) / 1.055, 2.4);
-      }
-    });
-
-    const L = 0.2126 * c[0] + 0.7152 * c[1] + 0.0722 * c[2];
-    return L > 0.179 ? "black !important" : "white !important";
+    return r * 0.299 + g * 0.587 + b * 0.114 > 186 ? "black !important" : "white !important";
```
Comment (Hinton) [inline src/app/modules/vault/modules/organization-badge/organization-name-badge.component.ts:53]: I think we  have two implementations for converting a string to color currently, with the avatar component. Any tech debt ticket to streamline this into a single implementation?

**Review by Hinton — APPROVED** [review]

File: src/app/modules/vault/modules/organization-badge/organization-name-badge.component.ts (hunk 2 above)
Comment (Hinton) [reply src/app/modules/vault/modules/organization-badge/organization-name-badge.component.ts:53]: Seems reasonable to add a note/subtask to that ticket

**Pull request merged**